modules/build/src/main/scala/scala/build/preprocessing/directives/RequirePlatformsDirectiveHandler.scala

```diff
@@ -29,4 +33,53 @@ case object RequirePlatformsDirectiveHandler extends RequireDirectiveHandler {
       case None =>
         None
     }
+
+  override def handleValues(
+    directive: StrictDirective,
+    path: Either[String, Path],
+    cwd: ScopePath
+  ): Either[BuildException, (Option[BuildRequirements], Seq[Scoped[BuildRequirements]])] = {
+    val values          = DirectiveUtil.stringValues(directive.values, path, cwd)
+    val nonscopedValues = values.filter(_._3.isEmpty)
+    val scopedValues    = values.filter(_._3.nonEmpty)
+    val nonScopedPlatforms = Option(nonscopedValues.map(v => Platform.normalize(v._1)))
+      .filter(_.nonEmpty)
+
+    val nonscoped =
+      nonScopedPlatforms.fold[Either[BuildException, Option[BuildRequirements]]](Right(None)) {
+        platforms =>
+          val parsed = Platform.parseSpec(platforms)
+          parsed.fold[Either[BuildException, Option[BuildRequirements]]](
+            Left(new MalformedPlatformError(platforms.mkString(", ")))
+          ) { p =>
+            Right(Some(BuildRequirements(
+              platform = Seq(BuildRequirements.PlatformRequirement(p))
+            )))
+          }
+      }
+    val scoped = scopedValues.groupBy(_._3).map {
+      case (Some(scopePath), list) =>
+        val platforms = list.map(_._1).map(Platform.normalize)
+        val parsed    = Platform.parseSpec(platforms)
+        parsed.fold[Either[BuildException, Seq[Scoped[BuildRequirements]]]](
+          Left(new MalformedPlatformError(platforms.mkString(", ")))
+        ) { p =>
+          Right(Seq(Scoped(
+            scopePath,
+            BuildRequirements(
+              platform = Seq(BuildRequirements.PlatformRequirement(p))
+            )
+          )))
+        }
+    }.foldLeft[Either[BuildException, Seq[Scoped[BuildRequirements]]]](Right(Seq.empty)) {
```

Comment:
Using
```scala
  .sequence
  .left.map(CompositeBuildException(_))
  .map(_.flatten)
```

(with `import scala.build.Ops._`) would be better, and would not discard errors (here, only one of the errors is retained)

Reply:
Wow, that is usefeul